src/frontend/context.rs

```diff
@@ -503,6 +503,19 @@ impl Context {
                 variables.free(num_outer_variables, builder);
                 builder.add_continue();
             }
+            ast::Statement::Return { value } => {
+                let value = match value {
+                    Some(value) => {
+                        match self.translate_term(value, false, exports, files, logger) {
+                            Ok(Term::Expression(value)) => value,
+                            _ => todo!(),
+                        }
+                    }
+                    None => todo!(),
+                };
+                variables.free(0, builder);
```

Comment:
The function have to call the destructor of every local variable before returning a value. But the evaluation of the return value must precede the destruction.